executor/filter.c

```diff
@@ -0,0 +1,113 @@
+#include "filter.h"
```

Comment:
Can we rename both files to `gamestate_filter.*` because filter is not descriptive